src/main/java/core/basesyntax/service/strategy/PurchaseFruitTransactionHandler.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseFruitTransactionHandler implements FruitTransactionHandler {
+    private final Storage storage;
+
+    public PurchaseFruitTransactionHandler(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        storage.delete(fruitTransaction.getFruit(), fruitTransaction.getQuantity());

```

storage.delete(...) may throw a RuntimeException for insufficient stock. Per the checklist guidance "Don't ignore exceptions in try-catch blocks" you should catch such exceptions (or let them propagate but wrap them) and rethrow a RuntimeException with an informative message that includes the transaction details and use the original exception as the cause. This makes failures easier to diagnose and satisfies the requirement to provide informative errors for invalid operations (e.g., purchase with insufficient balance).